src/library/vehicle.py

```diff
@@ -86,14 +107,35 @@ def winch_automation(self):
         if self.winch_enabled:
             arduino = None
 
+            #sets event to post winch_status as a constant rather than as a reference to winch_status
+            self.event.set()
             while arduino is None:
                 try:
                     arduino = ArduinoConnector(self)
                     print("[ALERT]    Rover & Winch     Arduino initialized")
                     self.winch_status = 1
                 except Exception as ex:
                     print("[ERROR]    Rover & Winch    ", ex)
-                time.sleep(1)
+
+                print("[TEST] event set, winch_status posting 1 at time=", datetime.now().strftime("%H:%M:%S"))
+
+                time.sleep(5)
+                break
+
+            self.event.clear()
+            print("[TEST] event cleared, winch_status posting 0", datetime.now().strftime("%H:%M:%S"))
+            time.sleep(5)
+
+            self.lock.acquire()
+            self.winch_status = 2
+            print("[TEST] lock aquired, winch_status posting 2", datetime.now().strftime("%H:%M:%S"))
+
+            self.lock.release()
+
+            time.sleep(5)
+            self.winch_status = 0
+            print("[TEST] lock released, winch_status posting 0", datetime.now().strftime("%H:%M:%S"))
+
```

Comment:
Are these lines meant to remain? They look like testing/debugging to me. If not, make sure to add back the `time.sleep(1)` to line 119.